### PR TITLE
add jubjub mul circuit and fix add

### PR DESCRIPTION
**Pull request by mdj33**

### CODE REVIEW

File: src/circuits/babyjub.rs [hunk 1]
```diff
@@ -197,22 +197,22 @@ impl<F: FieldExt> AltJubChip<F> {
...
-        let lambda = self.config.assign_line(region, &mut (), offset,
+        )?[2].clone();  // verify y1y2 is correct
+        let d_lambda = self.config.assign_line(region, &mut (), offset,
             [
                 Some(lambda1),
                 None,
                 None,
                 Some(lambda2),
-                Some(Limb::new(None, y1y2 * x1x2)),
+                Some(Limb::new(None, d*y1y2 * x1x2)),
```
Comment (xgaozoyoe) [inline src/circuits/babyjub.rs:207]: need to calculate d*y1y2*x1x2 in circuits, not in witness.

File: src/circuits/babyjub.rs [hunk 2]
```diff
@@ -221,29 +221,59 @@ impl<F: FieldExt> AltJubChip<F> {
...
             0
-        )?[4].clone();
+        )?[4].clone();   // gives x1y2 + x2y1
 
-        //x3 * (1+lambda) = x3s
-        let x3_f = x3s.value * (F::one() + lambda.value).invert().unwrap();
-        let x3 = self.config.assign_line(region, &mut (), offset,
+
+        //1+d*lambda
+        let x_d_lambda = F::one() + d_lambda.value;
```
Comment (xgaozoyoe) [inline src/circuits/babyjub.rs:230]: assign_constant, witness computation is not constraint.

File: src/circuits/babyjub.rs [hunk 3]
```diff
@@ -170,34 +163,41 @@ impl<F: FieldExt> AltJubChip<F> {
     ) -> Result<Point<F>, Error> {
         /* lambda = dx1x2y1y2
          * x3 = (x1y2 + y1x2)/(1 + lambda)
-         * y3 = (y1y2 - x1x2)/(1 - lambda)
+         * y3 = (y1y2 - ax1x2)/(1 - lambda)
          */
+        // constants
+        let a = F::from_str_vartime("21888242871839275222246405745257275088548364400416034343698204186575808495616").unwrap();
```
[the remark this answers is not in the document]
Comment (lanbones) [reply src/circuits/babyjub.rs:169]: At least we should move it to lazy_static

File: src/circuits/babyjub.rs (hunk 1 above)
Comment (lanbones) [reply src/circuits/babyjub.rs:207]: It's correct, but I hope this project can be format.

File: src/circuits/babyjub.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (xgaozoyoe) [reply src/circuits/babyjub.rs:169]: I will clean this after merge.

**Pull request merged**